java/runtime/src/main/java/io/ray/runtime/object/ObjectRefImpl.java

```diff
@@ -63,25 +63,34 @@ public void writeExternal(ObjectOutput out) throws IOException {
     out.writeObject(this.getId());
     out.writeObject(this.getType());
     RayRuntimeInternal runtime = (RayRuntimeInternal) Ray.internal();
-    byte[] ownerAddress = runtime.getObjectStore().getOwnershipInfo(this.getId());
-    out.writeInt(ownerAddress.length);
-    out.write(ownerAddress);
+    OwnershipInfo info = runtime.getObjectStore().getOwnershipInfo(this.getId());
+
+    out.writeInt(info.getSerializedOwnerAddress().length);
```

Comment:
Is it possible to make `OwnershipInfo` implements `Externalizable` and use `out.writeObject(info)` here?

Reply:
Good idea.